webtau-http/src/main/java/com/twosigma/webtau/http/datanode/StructuredDataNode.java

```diff
@@ -59,19 +59,66 @@ public DataNodeId id() {
     }
 
     @Override
-    public DataNode get(String name) {
+    public DataNode get(String nameOrPath) {
         if (isList()) {
-            return getAsCollectFromList(name);
+            return getAsCollectFromList(nameOrPath);
         }
 
+        int dotIdx = nameOrPath.indexOf('.');
+        if (dotIdx == -1) {
+            // simple name
+            return getChild(nameOrPath);
+        }
+
+        String rootName = nameOrPath.substring(0, dotIdx);
+        DataNode root = getChild(rootName);
+        String pathUnderRoot = nameOrPath.substring(dotIdx + 1);
+
+        return root.get(pathUnderRoot);
+    }
+
+    private DataNode getChild(String name) {
+        int openBraceIdx = name.indexOf('[');
+        int closeBraceIdx = name.indexOf(']');
+        if (openBraceIdx != -1 && closeBraceIdx != -1) {
+            return getIndexedChild(name, openBraceIdx, closeBraceIdx);
+        } else if (openBraceIdx != -1 || closeBraceIdx != -1) {
+            throw new IllegalArgumentException("Requested name " + name + " is not a simple name nor does it contain a properly formatted index");
+        }
+
+        // simple name
         return (children != null && children.containsKey(name)) ?
-                children.get(name):
+                children.get(name) :
                 new NullDataNode(id.child(name));
     }
 
+    private DataNode getIndexedChild(String name, int openBraceIdx, int closeBraceIdx) {
+        int additionalOpenIdx = name.indexOf('[', openBraceIdx + 1);
+        int additionalCloseId = name.indexOf(']', closeBraceIdx + 1);
+
+        if (additionalOpenIdx != -1 || additionalCloseId != -1 || openBraceIdx > closeBraceIdx) {
```

Comment:
what about cases like `[[1]`

Reply:
That'll throw.  That's what the `additional*Idx` checks are for, they look for brackets after the first one.

Reply:
doh